src/js/modules/Contact/ContactDetails.js

```diff
@@ -14,47 +14,35 @@ const i18nMessages = defineMessages({
   }
 });
 
-const ContactDetails = ({ contact, intl }) => {
-  const {name, phone, email, location, hours} = contact;
-
-  return (
-    <div className="coa-ContactDetails">
-      <SectionHeader isSerif={true}>{intl.formatMessage(i18nMessages.contactDetailsTitle)}</SectionHeader>
-      <div className="coa-ContactDetails__items">
-        { phone && (
-          <Phone
-            phone={phone.default}
-            ttyphone={phone.tty}
-          />
-        )}
-
-        { email && (
-          <Email email={email} />
-        )}
-
-        { location && (
-          <Address
-            name={name}
-            location={location}
-          />
-        )}
-
-        { hours && (
-          <Hours hours={hours} />
-        )}
-
-      </div>
-    </div>
-  );
-}
+const ContactDetails = ({ contact:{phone, email, location, hours}, intl }) => (
+  <div className="coa-ContactDetails">
+    <SectionHeader isSerif={true}>{intl.formatMessage(i18nMessages.contactDetailsTitle)}</SectionHeader>
+    { phone && (
+      <Phone
+        phone={phone}
+      />
+    )}
+
+    { email && (
+      <Email email={email} />
+    )}
+
+    { location && (
+      <Address location={location} />
+    )}
+
+    { hours && (
+      <Hours hours={hours} />
+    )}
+  </div>
+);
 
 ContactDetails.propTypes = {
   contact: PropTypes.shape({
-    phone: PropTypes.object,
-    email: PropTypes.string,
-    name: PropTypes.string,
-    location: PropTypes.object,
-    hours: PropTypes.array,
+    phone: PropTypes.Phone,
```

Comment:
simplified some proptypes so we could borrow child definitions like luqmaan showed us. That said, I'm not sure how to make phone optional unless it IS set. And then it should follow the PropTypes as defined by phone. I think this definition just requires phone, email, etc to all be set.

Reply:
is `PropTypes.Phone`, `PropTypes.Hours`, etc valid? I thought they needed to be standard types like string, boolean, func, etc in order for the type checking to work.
https://reactjs.org/docs/typechecking-with-proptypes.html

Reply:
Good call. I updated this so we could inherit the propTypes from the child components. Though, to do this I'm no longer specifying child props so now maybe the propTypes are less helpful? I'm open to suggestion on how to balance copy-pasting proptypes from composed components vs being explicit in our definitions.

Reply:
oh, i get it now. That makes sense.